sync/src/types.rs

```diff
@@ -838,41 +988,45 @@ impl SyncState {
     }
 }
 
-impl SyncSnapshot {
-    pub fn state(&self) -> &SyncState {
-        &self.state
+/** ActiveChain captures a point-in-time view of indexed chain of blocks. */
+#[derive(Clone)]
+pub struct ActiveChain {
+    shared: SyncShared,
+    snapshot: Arc<Snapshot>,
+    state: Arc<SyncState>,
+}
+
+impl ActiveChain {
+    pub fn store(&self) -> &ChainDB {
+        self.shared.store()
     }
 
-    pub fn get_block(&self, hash: &packed::Byte32) -> Option<core::BlockView> {
-        self.store.get_block(hash)
+    pub fn snapshot(&self) -> &Snapshot {
+        &self.snapshot
```

Comment:
It brings a potential risk of misusing the exposed `snapshot` and `store`.

What about keeping `snapshot` and `store` private and wrapping `pub fn get_block_hash`/`pub fn get_block_header`/`pub fn get_block` .